### PR TITLE
Chore: Upgrade Node to v22, update deps

### PR DESCRIPTION
**Pull request by drwpow**
## Changes

- Updates Node from `16` → `22`. Node 16 is EOL. Welcome to the future 😎 
- Updates deps. Because there were install errors for old packages on modern Node versions, and also because there were multiple security warnings.
- Updates GitHub Actions to current versions.

## Reviewing

- Look at [Preview Link](https://deploy-preview-242--dtcg-tr.netlify.app/)
- Look at GitHub logs

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -11,16 +11,17 @@ env:
...
-      - uses: actions/cache@v2
+      - uses: actions/checkout@v4
+      - uses: actions/cache@v4
         with:
           path: ~/.npm
           key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
           restore-keys: |
             ${{ runner.os }}-node-
-      - uses: actions/setup-node@v3
-        with: { node-version: 16, cache: npm }
```
Comment (drwpow) [inline .github/workflows/pr.yml:24]: From other PR: it’s generally safe these days to not specify the Node version. Major versions don’t break for most users anymore unless you’re using bleeding-edge/experimental features.

**Comment by kaelig** [comment]
I just updated the Netlify Site configuration to use Node v20. Node v22 should work too, but isn't fully supported.

**Comment by kaelig** [comment]
Looks like Chrome isn't available anymore – can you look into it?

**Comment by drwpow** [comment]
> Looks like Chrome isn't available anymore – can you look into it?

Sorry; what do you mean by Chrome not being available?

**Comment by kaelig** [comment]
Chrome must have been available in the old build image ([see error logs](https://app.netlify.com/sites/dtcg-tr/deploys/66c7920643f9a30008642d6b#L84-L96)), and isn't available with the Node 22 build image.

**Comment by drwpow** [comment]
> Chrome must have been available in the old build image ([see error logs](https://app.netlify.com/sites/dtcg-tr/deploys/66c7920643f9a30008642d6b#L84-L96)), and isn't available with the Node 22 build image. I believe it's necessary for ReSpec to build.

Just added a setup step for Puppeteer in Netlify. I think Netlify caches this well as it was pretty instant.

File: technical-reports/color/index.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
+<!doctype html>
```
Comment (drwpow) [inline technical-reports/color/index.html:1]: Just Prettier autoformatting in latest version

**Comment by drwpow** [comment]
> I just updated the Netlify Site configuration to use Node v20. Node v22 should work too, but isn't fully supported.

Also that’s fine! Node v20 is fine. I only updated to v22 because it’s been out for a while, it’s the latest stable version, and will be replacing v20 as the new active version in [only a couple months](https://nodejs.org/en/about/previous-releases). But it’s all fine, as long as we’re upgraded off an EOL version 😅

**Review by kaelig — APPROVED** [review]

**Pull request merged**